test/model/compose_test.dart

```diff
@@ -314,6 +314,28 @@ hello
         '#narrow/dm/1,2-dm/near/12345',
         '#narrow/pm-with/1,2-pm/near/12345');
     });
+
+    test('normalize links to always include a "/" after hostname', () {
+      void checkGeneratedLink({required String realmUrl, required String expectedLink}) {
+        final account = eg.selfAccount.copyWith(realmUrl: Uri.parse(realmUrl));
+        final store = eg.store(account: account);
```

Comment:
nit: just make a fresh Account from scratch:
```suggestion
        final store = eg.store(account: eg.account(realmUrl: Uri.parse(realmUrl)));
```

Or is there a reason that doesn't work?

Reply:
It would be:
```dart
      final store = eg.store(account: eg.account(user: eg.selfUser, realmUrl: Uri.parse(realmUrl)));
```
which is fine and actually as short as the other way:
```dart
      final store = eg.store(account: eg.selfAccount.copyWith(realmUrl: Uri.parse(realmUrl)));
```

Reply:
... Oh, I didn't see this reply when writing https://github.com/zulip/zulip-flutter/pull/1059#discussion_r1881120699 . I think I got got by the GitHub bug where a PR page often/usually live-updates with new comments, but then often doesn't, so I was looking at the page (in an existing tab) and didn't see any reply.

Anyway, if it needs that extra detail, no need to change it. And from your revision I see there's an existing test there that sets this up the same way.

Reply:
I guess the non-copy-with implementation would be a tiny bit more realistic because we supposedly (there is an TODO on `eg.account`) will regenerate a bunch of fields from scratch as well.

Reply:
Hmm true. Not a bunch, I don't think; but the account ID, and maybe the API key too. Sure.